src/wasm.rs

```diff
@@ -140,22 +140,146 @@ impl Server {
     pub fn register_tag_values_callback(&mut self, _f: Function) {}
 }
 
-/// Parses a string representation of a json file and returns the corresponding JsValue representation
+/// Parse flux into an AST representation. The AST will be generated regardless
+/// of valid flux. As such, no error handling is needed.
 #[wasm_bindgen]
-pub fn parse(s: &str) -> JsValue {
-    let mut p = Parser::new(s);
-    let file = p.parse_file(String::from(""));
+pub fn parse(script: &str) -> JsValue {
+    let mut parser = Parser::new(script);
+    let parsed = parser.parse_file("".to_string());
 
-    JsValue::from_serde(&file).unwrap()
+    JsValue::from_serde(&parsed).unwrap()
 }
 
-/// Format a JS file.
+/// Attempt to format flux. If the flux is invalid, just return the
+/// original text.
 #[wasm_bindgen]
-pub fn format_from_js_file(js_file: JsValue) -> String {
-    if let Ok(file) = js_file.into_serde::<File>() {
-        if let Ok(converted) = convert_to_string(&file) {
-            return converted;
+pub fn format(script: &str) -> String {
+    let mut parser = Parser::new(script);
+    let ast = parser.parse_file("".to_string());
+
+    match convert_to_string(&ast) {
+        Ok(formatted) => {
+            if formatted.contains("<INVALID_OP>") {
```

Comment:
This doesn't seem like a reliable way to know if a script was formatted successfully. For example there could be a string literal in the program that contains this text, or perhaps there could scenarios where parsing fails but this particular text is not generated.

Ideally there would be an easy way of discovering errors provided by `flux-core`. It looks like walking the AST and looking for `BadExpr` or `BadStmt` nodes would tell you if you if parsing resulted in any errors. You could perhaps do that here until a better fix is available.

Reply:
Yeah, I thought about that. I should have changed this PR to "draft" because I'm not even sure if this function is going to get used. I'd like to take a holistic view of what we export in the first place. I like your idea, I'm just not quite sure this patch is going to get merged in this state.